### PR TITLE
Make sonata-project/notification-bundle optional

### LINKED ISSUE #1238: Remove coupling to NotificationBundle
## Feature Request

This is the last bundle that relies on the [SonataNotificationBundle](https://github.com/sonata-project/SonataNotificationBundle/) for no good reason. The original idea was to allow async snapshot creation / cleanup. This can also be done via commands and shouldn't take too long for a normal number of pages.

To allow deprecating the usage, we could introduce a new `SnapshotManagerInterface` with two methods `createSnapshotsForSite` and `cleanupSnashotsForSite`.

WDYT @sonata-project/contributors

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this feature is BC.

Closes #1238

## Changelog

```markdown
### Changed
- Publishing is synchronous by default
- Made `sonata-project/notification-bundle` optional
```

## To do

- [ ] Update the tests;
- [x] Update the documentation;
- [x] Add an upgrade note.

### CODE REVIEW

File: src/Command/CleanupSnapshotsCommand.php [hunk 1]
```diff
@@ -68,11 +68,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
                     $output->write(sprintf('<info>%s</info> - Cleaning up snapshots ...', $site->getName()));
                 }
 
-                $this->getNotificationBackend($input->getOption('mode'))->createAndPublish('sonata.page.cleanup_snapshots', [
-                    'siteId' => $site->getId(),
-                    'mode' => $input->getOption('mode'),
-                    'keepSnapshots' => $input->getOption('keep-snapshots'),
-                ]);
+                if ('asnc' === $input->getOption('mode')) {
```
Comment (OskarStark) [inline src/Command/CleanupSnapshotsCommand.php:71]:
```suggestion
                if ('async' === $input->getOption('mode')) {
```

File: src/Command/CreateSnapshotsCommand.php [hunk 2]
```diff
@@ -58,10 +58,20 @@ public function execute(InputInterface $input, OutputInterface $output)
                     $output->write(sprintf('<info>%s</info> - Generating snapshots ...', $site->getName()));
                 }
 
-                $this->getNotificationBackend($input->getOption('mode'))->createAndPublish('sonata.page.create_snapshots', [
-                    'siteId' => $site->getId(),
-                    'mode' => $input->getOption('mode'),
-                ]);
+                if ('asnc' === $input->getOption('mode')) {
```
Comment (OskarStark) [inline src/Command/CreateSnapshotsCommand.php:61]:
```suggestion
                if ('async' === $input->getOption('mode')) {
```

File: src/Publisher/SimplePublisher.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+namespace Sonata\PageBundle\Publisher;
+
+use Sonata\PageBundle\Model\PageInterface;
+use Sonata\PageBundle\Model\PageManagerInterface;
+use Sonata\PageBundle\Model\SiteInterface;
+use Sonata\PageBundle\Model\SnapshotManagerInterface;
+use Sonata\PageBundle\Model\TransformerInterface;
+
+final class SimplePublisher implements Publisher
```
Comment (OskarStark) [inline src/Publisher/SimplePublisher.php:22]: Please add @author to the new classes, thanks

**Comment by VincentLanglet** [comment]
Do you plan/have time to finish this @core23 ? :)

**Comment by core23** [comment]
I lost interest for this bundle. So I haven't time to finish this :(

**Pull request closed without merging**